R/tar-terra-rast.R

```diff
@@ -175,6 +182,38 @@ tar_rast_write <- function(filetype, gdal, preserve_metadata) {
             #move the zip file to the expected place
             file.rename(file.path(tmp, basename(path)), path)
         },
+        gdalraster_sozip = function(object, path) {
+
+            tmp <- withr::local_tempdir()
+
+            dir.create(file.path(tmp, dirname(path)), recursive = TRUE)
+
+            terra::writeRaster(
+                object,
+                file.path(tmp, path),
+                filetype = filetype,
+                overwrite = TRUE,
+                gdal = gdal
+            )
+
+            raster_files <- list.files(file.path(tmp, dirname(path)), full.names = TRUE)
+
+            # create seek-optimized zip file using gdalraster
+            gdalraster::addFilesInZip(
+                path,
+                raster_files,
+                full_paths = FALSE,
+                overwrite = TRUE,
+                sozip_enabled = "YES",
+                num_threads = 1,
+                quiet = TRUE
+            )
+            # always create sozip regardless of file size (sozip_enabled = "YES")
+            # TODO: allow user control of number of threads?
+            #       how does num_threads interact multiple workers etc.?
+
+            unlink(file.path(tmp, path), )
```

Comment:
should unlink have a trailing comma here?

Reply:
Thanks for catching, I believe I spotted this in the merge commit!